### DISCUSSION TITLE
Status on React Component Collab Support

### DISCUSSION THREAD

**Discussion by yaralahruthik** [excalidraw/excalidraw#4924] [Q&A] [answered]
From what I see currently, the excalidraw-room does not work locally for me, the server is up yet excalidraw does not connect.
I did try to have my own implementation of server, but the onchange seems to fire always and it puts me in an infinite loop.

Could anyone help me out here? Anything I am doing wrong?

**Comment by ad1992** [comment, marked as answer]
If you want to use the `excalidraw-room` you will have to update the server end point as well, more details in https://github.com/excalidraw/excalidraw/discussions/3879#discussioncomment-1110524

As per `onChange`, right now it gets triggered for every single change so to optimize it you can cache the [`scene version`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#getsceneversion) and compare the existing and new scene version before syncing the data, this would make sure you sync the data only when its needed.

**Reply by Favourz1** [reply]
How can I get the current sceneData that can be passed as an argument in excalidrawAPI.updateScene(sceneData);

**Context:** I'm building my own collaboration feature and just want to cache and send/update the current scene data myself. So far I only see way to update it and no way to get the current scene data.

cc: @ad1992
